### PR TITLE
[SandboxIR] Add the ExtractElementInst class

### PR DESCRIPTION
**Pull request by slackito**

### CODE REVIEW

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 1]
```diff
@@ -631,6 +631,48 @@ define void @foo(i1 %c0, i8 %v0, i8 %v1, i1 %c1) {
...
+}
+)IR");
+  Function &LLVMF = *M->getFunction("foo");
+  sandboxir::Context Ctx(C);
+  auto &F = *Ctx.createFunction(&LLVMF);
+  auto *ArgVec = F.getArg(0);
+  auto *ArgIdx = F.getArg(1);
+  auto *BB = &*F.begin();
+  auto It = BB->begin();
+  auto *Ins0 = cast<sandboxir::ExtractElementInst>(&*It++);
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:648]: Rename `Ins0` to `Extr` or `EI`

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 2]
```diff
@@ -631,6 +631,48 @@ define void @foo(i1 %c0, i8 %v0, i8 %v1, i1 %c1) {
...
+  auto *Ret = &*It++;
+
+  EXPECT_EQ(Ins0->getOpcode(), sandboxir::Instruction::Opcode::ExtractElement);
+  EXPECT_EQ(Ins0->getOperand(0), ArgVec);
+  EXPECT_EQ(Ins0->getOperand(1), ArgIdx);
+  EXPECT_EQ(Ins0->getVectorOperand(), ArgVec);
+  EXPECT_EQ(Ins0->getIndexOperand(), ArgIdx);
+  auto *NewI1 =
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:656]: I think we need a test for `getVectorOperandType()`. Something like:
```
EXPECT_EQ(EI->getVectorOperandType(), ArgVec->getType());
```

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 1 above)
Comment (slackito) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp]: Done.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 2 above)
Comment (slackito) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp:658]: Done!

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 3]
```diff
@@ -631,6 +631,50 @@ define void @foo(i1 %c0, i8 %v0, i8 %v1, i1 %c1) {
...
+  EXPECT_EQ(EI->getOpcode(), sandboxir::Instruction::Opcode::ExtractElement);
+  EXPECT_EQ(EI->getOperand(0), ArgVec);
+  EXPECT_EQ(EI->getOperand(1), ArgIdx);
+  EXPECT_EQ(EI->getVectorOperand(), ArgVec);
+  EXPECT_EQ(EI->getIndexOperand(), ArgIdx);
+  EXPECT_EQ(EI->getVectorOperandType(), ArgVec->getType());
+
+  auto *NewI1 =
+      cast<sandboxir::ExtractElementInst>(sandboxir::ExtractElementInst::create(
+          ArgVec, ArgIdx, Ret, Ctx, "NewInsBeforeRet"));
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:660]: nit: "NewInsBeforeRet" -> "NewExtrBeforeRet"

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 4]
```diff
@@ -631,6 +631,50 @@ define void @foo(i1 %c0, i8 %v0, i8 %v1, i1 %c1) {
...
+  EXPECT_EQ(NewI1->getOperand(0), ArgVec);
+  EXPECT_EQ(NewI1->getOperand(1), ArgIdx);
+  EXPECT_EQ(NewI1->getNextNode(), Ret);
+
+  auto *NewI2 =
+      cast<sandboxir::ExtractElementInst>(sandboxir::ExtractElementInst::create(
+          ArgVec, ArgIdx, BB, Ctx, "NewInsAtEndOfBB"));
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:667]: "NewInsAtEndOfBB" -> "NewExtrAtEndOfBB"

**Review by vporpo — APPROVED** [review]
Just a couple of minor comments, looks good otherwise. Thanks!

**Comment by slackito** [comment]
The windows build failure seems unrelated: test\tools\llvm-objdump\ELF\RISCV\Output\source-interleave.ll. I'm going to merge this now.

**Pull request merged**